homeassistant/components/climacell/const.py

```diff
@@ -155,165 +149,182 @@
 CC_ATTR_CLOUD_BASE = "cloudBase"
 CC_ATTR_CLOUD_CEILING = "cloudCeiling"
 
-CC_SENSOR_TYPES = [
-    {
-        ATTR_FIELD: CC_ATTR_FEELS_LIKE,
-        ATTR_NAME: "Feels Like",
-        CONF_UNIT_SYSTEM_IMPERIAL: TEMP_FAHRENHEIT,
-        CONF_UNIT_SYSTEM_METRIC: TEMP_CELSIUS,
-        ATTR_METRIC_CONVERSION: lambda val: temp_convert(
-            val, TEMP_FAHRENHEIT, TEMP_CELSIUS
-        ),
-        ATTR_IS_METRIC_CHECK: True,
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
-    },
-    {
-        ATTR_FIELD: CC_ATTR_DEW_POINT,
-        ATTR_NAME: "Dew Point",
-        CONF_UNIT_SYSTEM_IMPERIAL: TEMP_FAHRENHEIT,
-        CONF_UNIT_SYSTEM_METRIC: TEMP_CELSIUS,
-        ATTR_METRIC_CONVERSION: lambda val: temp_convert(
-            val, TEMP_FAHRENHEIT, TEMP_CELSIUS
-        ),
-        ATTR_IS_METRIC_CHECK: True,
-        ATTR_DEVICE_CLASS: DEVICE_CLASS_TEMPERATURE,
-    },
-    {
-        ATTR_FIELD: CC_ATTR_PRESSURE_SURFACE_LEVEL,
-        ATTR_NAME: "Pressure (Surface Level)",
-        CONF_UNIT_SYSTEM_IMPERIAL: PRESSURE_INHG,
-        CONF_UNIT_SYSTEM_METRIC: PRESSURE_HPA,
-        ATTR_METRIC_CONVERSION: lambda val: pressure_convert(
+
+@dataclass
+class ClimaCellSensorMetadata:
+    """Metadata about an individual ClimaCell sensor."""
+
+    field: str
+    name: str
+    unit_imperial: str | None = None
+    unit_metric: str | None = None
+    metric_conversion: Callable | float = 1.0
```

Comment:
`Callable` could have been more specific

Reply:
what's a good alternative type for this?

Reply:
Probably this one here:
```py
    metric_conversion: Callable[[float], float] | float = 1.0 | float = 1.0
```